datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -600,6 +603,14 @@ def check(self, instance):
 
     def run(self):
         try:
+            while self.check_initializations:
+                initialization = self.check_initializations.popleft()
+                try:
+                    initialization()
+                except Exception:
+                    self.check_initializations.appendleft(initialization)
+                    raise
```

Comment:
Seems this would have a similar effect:

```
                self.check_initializations[0]()
                self.check_initializations.pop(0)
```